### PR TITLE
Update model tags & properties, and minor fixes

### PR DESCRIPTION
**Pull request by novaturient95**

### CODE REVIEW

**Comment by novaturient95** [comment]
Wait till list of tags are properties are updated.

File: training/model_management/environments/model-management/context/requirements.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ azure-core=={{latest-pypi-version}}
 azure-ai-ml=={{latest-pypi-version}}
 azure-identity=={{latest-pypi-version}}
 azureml-evaluate-mlflow=={{latest-pypi-version}}
-ffmpeg=={{latest-pypi-version}}
+ffmpeg-python=={{latest-pypi-version}}
```
Comment (skasturi) [inline training/model_management/environments/model-management/context/requirements.txt:5]: Is this not a dependency of azureml-evaluate-mlflow that supports whisper?

Comment (novaturient95) [reply training/model_management/environments/model-management/context/requirements.txt:5]: ffmpeg-python is dependency for whisper predict script

**Comment by skasturi** [comment]
> datasets==2.3.2

Remove this as it's included as part evaluate-mlflow

---
Refers to: training/model_management/environments/model-management/context/requirements.txt:10 in 4c6ed97. [](commit_id = 4c6ed975ee784d9163872ac6737808c53fe21cb0, deletion_comment = False)

File: training/model_management/environments/model-management/context/requirements.txt (hunk 1 above)
Comment (skasturi) [reply training/model_management/environments/model-management/context/requirements.txt:5]: Ideally, this needs to become a dependency of evaluate-mlflow.

File: training/model_management/environments/model-management/context/requirements.txt [hunk 2]
```diff
@@ -2,10 +2,6 @@ azure-core=={{latest-pypi-version}}
...
 langcodes=={{latest-pypi-version}}
-protobuf==3.20.0
 sentencepiece==0.1.97
```
Comment (skasturi) [inline training/model_management/environments/model-management/context/requirements.txt:7]: Could you please point me to where this being used in the code?

File: training/model_management/src/azureml/model/mgmt/downloader/downloader.py [hunk 3]
```diff
@@ -48,22 +52,23 @@ def model_info(self) -> ModelInfo:
...
         datasets = []
         for tag in all_tags:
-            if langcodes.tag_is_valid(tag):
-                supported_langs.append(tag)
+            if langcodes.tag_is_valid(tag) and tag not in LANGUAGE_CODE_EXCEPTIONS:
+                languages.append(tag)
             elif tag.startswith("dataset:"):
                 datasets.append(tag.split(":")[1])
             elif tag.startswith("license:"):
                 props["license"] = tag.split(":")[1]
```
Comment (skasturi) [inline training/model_management/src/azureml/model/mgmt/downloader/downloader.py:67]: (nit) add new line

File: training/model_management/environments/model-management/context/requirements.txt (hunk 2 above)
Comment (novaturient95) [reply training/model_management/environments/model-management/context/requirements.txt:7]: used when loading tokenizer for some of the models (Jean-Baptiste-camembert-ner)

File: training/model_management/environments/model-management/context/requirements.txt (hunk 1 above)
Comment (novaturient95) [reply training/model_management/environments/model-management/context/requirements.txt:5]: Created a bug for this

File: training/model_management/src/azureml/model/mgmt/downloader/downloader.py [hunk 4]
```diff
@@ -48,22 +52,23 @@ def model_info(self) -> ModelInfo:
...
-        props["dataset"] = ", ".join(datasets)
-        props["language"] = ", ".join(supported_langs)
+        if datasets:
+            props["datasets"] = ", ".join(datasets)
+        if languages:
+            props["languages"] = ", ".join(languages)
         return props
```
[the remark this answers is not in the document]
Comment (novaturient95) [reply training/model_management/src/azureml/model/mgmt/downloader/downloader.py:84]: Done

File: training/model_management/src/azureml/model/mgmt/downloader/downloader.py (hunk 3 above)
Comment (novaturient95) [reply training/model_management/src/azureml/model/mgmt/downloader/downloader.py:77]: Done

**Review by skasturi — APPROVED** [review]

**Pull request merged**